### PR TITLE
Fix versioning

### PR DESCRIPTION
**Pull request by PabstMirror**
Fix versioning system
`setVariable` doesn't seem to work on local variables.

### CODE REVIEW

**Comment by commy2** [comment]
:+1:

**Comment by PabstMirror** [comment]
```
missionNamespace setVariable ["_dependencyIsPresent", _dependencyInfo select 2];
if (_dependencyIsPresent) then {
```
Old code was doing this, as far as I know we still need the call compile to convert the string into a bool.

**Comment by commy2** [comment]
>I believe there is another method other than call compile, right?

`missionNamespace getVariable` can not replace all `call compile`. It only works when you have the name of a (global) variable.
`missionNamespace getVariable "CBA_Gvar" == call compile "CBA_Gvar" == CBA_Gvar`

It does not work when you compile a whole expression (like here) instead of just a var name.

**Comment by NouberNou** [comment]
OK I think I need a better explanation of what those lines are doing, because I am confused as hell... what is `_dependencyInfo select 2`?

**Comment by PabstMirror** [comment]
@NouberNou 
`_dependecyInfo` should be something like
```
["cba_main_a3",[1,0,0],"isClass(configFile >> 'CfgPatches' >> 'A3_Map_Stratis')"]
//but someone could put "true" which could mistakenly compile to `1` and get
["cba_xeh",[1,0,0],1]],["CBA",["cba_main",[1,0,0],1]
```

@ViperMaul `setVariable` won't work here because we need to convert the conditional string
```
"isClass(configFile >> 'CfgPatches' >> 'A3_Map_Stratis')"
```
into a boolean for the `if`, setVariable would just store the raw string into the var, it needs to be evaluated with `call`

Tested with a dependency of `XEH[] = {"cba_xeh", {6,0,0}, 1};` which was handled without script error, but will write warning to rpt.

File: addons/versioning/XEH_postInit.sqf [hunk 1]
```diff
@@ -26,7 +26,12 @@ if (!SLX_XEH_DisableLogging) then {
         _mod = _x select 0;
         _dependencyInfo = _x select 1;
         _class = (configFile >> "CfgPatches" >> (_dependencyInfo select 0));
-        missionNamespace setVariable ["_dependencyIsPresent", _dependencyInfo select 2];
+        private _dependencyIsPresent = call compile format ["%1", (_dependencyInfo select 2)];
+        if ((isNil "_dependencyIsPresent") || {!(_dependencyIsPresent isEqualType false)}) then {
+            //https://dev.withsix.com/issues/74516 - The code could return non-bool, if "true" is converted to "1" durring binerization
```
Comment (Killswitch00) [inline addons/versioning/XEH_postInit.sqf]: binarization

File: addons/versioning/XEH_postInit.sqf [hunk 2]
```diff
@@ -26,7 +26,12 @@ if (!SLX_XEH_DisableLogging) then {
         _mod = _x select 0;
         _dependencyInfo = _x select 1;
         _class = (configFile >> "CfgPatches" >> (_dependencyInfo select 0));
-        missionNamespace setVariable ["_dependencyIsPresent", _dependencyInfo select 2];
+        private _dependencyIsPresent = call compile format ["%1", (_dependencyInfo select 2)];
+        if ((isNil "_dependencyIsPresent") || {!(_dependencyIsPresent isEqualType false)}) then {
+            //https://dev.withsix.com/issues/74516 - The code could return non-bool, if "true" is converted to "1" durring binerization
+            WARNING("Versioning Conditional return is bad" + str _x);
```
Comment (Killswitch00) [inline addons/versioning/XEH_postInit.sqf]: conditional

**Pull request merged**